nettests/nettests.go

```diff
@@ -50,6 +51,12 @@ type Controller struct {
 	msmts       map[int64]*database.Measurement
 	msmtPath    string          // XXX maybe we can drop this and just use a temporary file
 	inputIdxMap map[int64]int64 // Used to map mk idx to database id
+
+	// numInputs is the total number of inputs
+	numInputs int
+
+	// curInputIdx is the current input index
+	curInputIdx int
```

Comment:
Why don't you define these two as int64 so that the int size is not platform dependent and you can avoid casting it to `int64` when passing it to `inputIdxMap`?

Reply:
That's a good question. The direct answer is that Go has a notion of integer that depends on the native size of registers on the platform. This is reasonable because `int64` operations are expensive when registers are actually `32` bit wide.

Because of this performance improving design choice, Go uses `int` as index. Therefore, I find it mostly natural that the code uses `int` as index.

Unless there is a way to force to to return an `int64` index when `range`-ing over a slice, I do not think it is wise to switch these values to `int64` because we would anyway need to case the loop index to an int64.

I believe the reason why we're using `int64` here is mostly caused by the fact that MK was using a `int64` to represent indices in JSON messages. But that's a different story because JSON defines the integer as `int53`, so there was no choice but define that field as being 64 bit wide.

However, now that we're integrating more and more Go code with Go code, keeping the 64 bit indexing structure seems to me a legacy implementation detail.

If Go on a specific platform cannot handle offsets larger than 32 bit, there is no point for our Go code interfacing with Go code to cast to 64 bit.

There may however be specific database reasoning of which I'm not aware that may suggest to use 64 bits integer for representing offsets in the database.

If that is the case, then I'd recommend to refactor the code such that we perform the cast to `int64` when we're as close as possible to the database related code, and just use `int` in all the rest of the code to avoid unnecessary casts.

Otherwise, and especially because we've not made any public release yet, I would say that we would be better off by just always using `int`, including in the database.

This seems to me the best way to address this issue. (Incidentally, this is consistent with the Google code style that advocates for using the naturally sized integers on any platform, unless there are specific needs to user a specific integer size, i.e. for binary protocols - since Go originated at Google, I can see how this recommendation transpired into the language and I'd argue that following along is the idiomatic way to go in this case.)

Reply:
Then maybe we should change to int64 in the index of `inputIdxMap`?

Reply:
Do you mean changing to `int`? If so, yes, I was considering on doing that. But I did not perform this action as part of this PR, as the diff was quite messy already.

Is it fine if I create an issue for that?

Reply:
Sure let's make an issue for it.

Reply:
See #48